#162170720 pagination supports for article

okoroemeka opened the pull request:
#### What does this PR do?
Have pagination on Authors-haven

#### Description of Task to be completed?
Have the following endpoint working
`GET api/article`

#### How should this be manually tested?
After cloning the Repo, cd into it and run `npm install` after that run `npm run dev:start`.
* Using Postman test the route.

#### What are the relevant pivotal tracker stories?
[#162170720](https://www.pivotaltracker.com/story/show/162170720)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-12-11 at 6 01 40 pm" src="https://user-images.githubusercontent.com/32361413/49816829-06ad0680-fd6f-11e8-9389-5c1aa4087a5d.png">

marcdomain reviewed (changes requested):
Kindly address the following:
- This PR title does not conform to the previous ones.
- Squash your commits.
- I think the route should be in this format "**/api/articles?page=**"

Commit:
[Feat #162170720] Add pagination to project

- Add controller for pagination

- Add routes for pagination

- Add test for pagination controller

- Add documentation
[Finishes #162170720]

okoroemeka commented:
@marcdomain thanks for pointing those things out, I have implemented those changes.
